### PR TITLE
[Visualization utils] Add visualization utils for plotting images(plain, with bounding boxes and segmentation masks)

### PR DESCRIPTION
**Pull request by sineeli**
This PR adds the following export utils for visualization of images

1. `draw_bounding_boxes` -  draw bounding boxes on give images.
2. `draw_segmentation_masks` - draw masks on given images.
3. `plot_image_gallery` - plots given images
4. `plot_bounding_box_gallery` - plots images with bounding boxes(y_true and y_pred).
5. `plot_segmentation_mask_gallery` - plots images with masks(y_true and y_pred).

Reference: https://colab.research.google.com/gist/sineeli/8b2caa962f0c33d9450a4e5e70ec2d43/-keras3-visualization-utils.ipynb

### CODE REVIEW

File: keras/src/visualization/draw_bounding_boxes.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+                label = class_mapping[c]
+                if confidences is not None:
+                    conf = confidences[i][box_i]
+                    label = f"{label} | {conf:.2f}"
+
+                font_x1, font_y1 = _find_text_location(
+                    x1, y1, font_scale, text_thickness
+                )
+                cv2.putText(
+                    _image,
```
Comment (fchollet) [inline keras/src/visualization/draw_bounding_boxes.py:153]: Please use keyword arguments for better readability

File: keras/src/visualization/plot_image_gallery.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    dpi=60,
+    legend_handles=None,
+    data_format=None,
+):
+    """Displays a gallery of images.
+
+    Args:
+        images: A 4D tensor or NumPy array of images. Shape should be
+           `(batch_size, height, width, channels)`.
+        value_range: A tuple specifying the value range of the images
```
Comment (fchollet) [inline keras/src/visualization/plot_image_gallery.py:63]: Require this to be specified is a bit user-unfriendly. I would recommend defaulting to (0, 255), which is what image plotting utilities normally expect.

File: keras/src/visualization/plot_segmentation_mask_gallery.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+from keras.src.visualization.draw_segmentation_masks import (
+    draw_segmentation_masks,
+)
+from keras.src.visualization.plot_image_gallery import plot_image_gallery
+
+
+@keras_export("keras.visualization.plot_segmentation_mask_gallery")
+def plot_segmentation_mask_gallery(
+    images,
+    value_range,
```
Comment (fchollet) [inline keras/src/visualization/plot_segmentation_mask_gallery.py:17]: Same here

File: keras/src/visualization/draw_bounding_boxes.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    import cv2
+except ImportError:
+    cv2 = None
+
+
+@keras_export("keras.visualization.draw_bounding_boxes")
+def draw_bounding_boxes(
+    images,
+    bounding_boxes,
+    color,
```
Comment (fchollet) [inline keras/src/visualization/draw_bounding_boxes.py:20]: This should be have a default value (and thus come after the required args)

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/src/visualization/draw_bounding_boxes.py (hunk 1 above)
Comment (sineeli) [reply keras/src/visualization/draw_bounding_boxes.py:153]: Done

File: keras/src/visualization/plot_image_gallery.py (hunk 2 above)
Comment (sineeli) [reply keras/src/visualization/plot_image_gallery.py:63]: Yup make sense. Done!

File: keras/src/visualization/draw_bounding_boxes.py (hunk 4 above)
Comment (sineeli) [reply keras/src/visualization/draw_bounding_boxes.py:20]: Done!

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Comment by miticollo** [comment]
@sineeli I have a question. Consider https://github.com/keras-team/keras/pull/20401/files#diff-c8b7d21a681718ca171004545a12d8104c19013b18075a13b8273125927448b3R105 , it says that `plot_image_gallery` also accepts a single image, namely 3D tensor. But if you pass a single image a `ValueError` is raised by `_extract_image_batch`. Could it be a good idea to add
```python
    if batch_size == 1: 
        keras.ops.expand_dims(images, 0)
```
after `batch_size = ops.shape(images)[0] if len(ops.shape(images)) == 4 else 1`?
Another solution could it be move `if len(ops.shape(images)) != 4:` from `_extract_image_batch` into `plot_image_gallery` at beginning. This latter solution doesn't require to change function docstring.

File: keras/src/visualization/plot_image_gallery.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    show = show if show is not None else (path is None)
+    data_format = data_format or backend.image_data_format()
+
+    batch_size = ops.shape(images)[0] if len(ops.shape(images)) == 4 else 1
+
+    rows = rows or int(math.ceil(math.sqrt(batch_size)))
+    cols = cols or int(math.ceil(batch_size // rows))
+    num_images = rows * cols
+
+    images = _extract_image_batch(images, num_images, batch_size)
```
Comment (sineeli) [inline keras/src/visualization/plot_image_gallery.py:111]: Hey @miticollo,

Thanks for pointing out this, I think we can remove this `raise ValueError` from `_extract_image_batch` and as per your suggestion we can expand dimension when image is just 3D tensor.

```python3
if len(ops.shape(images)) == 3: 
  images  = ops.expand_dims(images, 0)
batch_size = ops.shape(images)[0]
```

And the rest of the logic should hold good.
